### PR TITLE
doc: add instructions on how to release the package

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**

### CODE REVIEW

**Comment by Meulengracht** [comment]
LGTM. Thank you for this.

**Review by stulluk — APPROVED** [review]
LGTM. Thank you so much.

File: README.md [hunk 1]
```diff
@@ -27,11 +27,49 @@ TODO: Write documentation for how to build initrd locally
...
+following these steps:
 
-TODO: Write documentation for how to integrate initrd.img to your (custom or canonical) kernel 
+1. Update the changelog with latest changes since last release (use `dch -i` for this)
+1. Propose a PR to the repo with the new changelog, get it reviewed and merged
+1. Tag the repository with the new version
+1. Build the source package by running
+
+        git clean -fdx
+        dpkg-buildpackage -S -sa -d
```
Comment (mvo5) [inline README.md:45]: This should probably converted to "git buildpackage" at some point, it will ensure the "git clean" step is not needed and that it's always build in a clean tree.

File: README.md [hunk 2]
```diff
@@ -27,11 +27,49 @@ TODO: Write documentation for how to build initrd locally
...
+
+1. Compare with the latest package that was uploaded to the snappy-dev
+PPA to make sure that the changes are correct.  For this, you can
+download the .dsc file and the tarball from the PPA, then run debdiff
+to find out the differences:
+
+        wget https://launchpad.net/~snappy-dev/+archive/ubuntu/image/+sourcefiles/ubuntu-core-initramfs/<old_version>/ubuntu-core-initramfs_<old_version>.dsc \
```
Comment (mvo5) [inline README.md:52]: you can use  `dget` here on the dsc file, that will automatically fetch the tar.xz

**Review by mvo5 — COMMENTED** [review]
Thanks, this looks nice. Two small suggestions/notes inline

File: README.md (hunk 2 above)
Comment (alfonsosanchezbeato) [reply README.md]: Ah, nice one, changed now

File: README.md (hunk 1 above)
Comment (alfonsosanchezbeato) [reply README.md]: Changed to `gbp buildpackage` (you meant that one, right?). But it looks that cleaning up is still needed, although gbp now errors if the tree is not clean, which is nice.

**Review by mvo5 — APPROVED** [review]
Thanks for this!

File: README.md [hunk 3]
```diff
@@ -27,11 +27,48 @@ TODO: Write documentation for how to build initrd locally
...
+PPA](https://launchpad.net/~snappy-dev/+archive/ubuntu/image) by
+following these steps:
 
-TODO: Write documentation for how to integrate initrd.img to your (custom or canonical) kernel 
+1. Update the changelog with latest changes since last release (use `dch -i` for this)
+1. Propose a PR to the repo with the new changelog, get it reviewed and merged
+1. Tag the repository with the new version
+1. Build the source package by running
+
+        git clean -fdx
```
Comment (xnox) [inline README.md:44]:
```suggestion
        git clean -ffdx
```
Specify a second f flag, to force delete git repositories nested under vendor/gopath

Otherwise vendor/gopath is not deleted, and then not recreated by gbp clean, and potentially old/incorrect git commit is used for the custom snap-bootstrap which is used by the cloudimg feature.

File: README.md [hunk 4]
```diff
@@ -27,11 +27,48 @@ TODO: Write documentation for how to build initrd locally
...
+        gbp buildpackage -S -sa -d
```
Comment (xnox) [inline README.md:45]:
```suggestion
        gbp buildpackage -S -sa -d --git-ignore-branch
```

Currently we use branch names that do not match gbp expectations, hence pass the flag to ignore the branch name.

File: README.md [hunk 5]
```diff
@@ -27,11 +27,48 @@ TODO: Write documentation for how to build initrd locally
...
-When your initrd DEB package is built, you need to perform following steps to make it a part of kernel snap:
+The UC initrd is included in kernel snaps. However, the process to get
+it inside the kernel snap is not immediate and consists of a few
+steps. First, we need to build the `ubuntu-core-initramfs`
+debian package in the [snappy-dev/image
+PPA](https://launchpad.net/~snappy-dev/+archive/ubuntu/image) by
+following these steps:
 
-TODO: Write documentation for how to integrate initrd.img to your (custom or canonical) kernel 
+1. Update the changelog with latest changes since last release (use `dch -i` for this)
```
Comment (xnox) [inline README.md:39]:
```suggestion
1. Update the changelog with latest changes since last release (use `dch -i` for this)
1. Commit using `debcommit --release -a` which will commit the changelog update & tag the release
```

File: README.md [hunk 6]
```diff
@@ -27,11 +27,48 @@ TODO: Write documentation for how to build initrd locally
...
+1. Propose a PR to the repo with the new changelog, get it reviewed and merged
+1. Tag the repository with the new version
```
Comment (xnox) [inline README.md:41]:
```suggestion
1. Push the tag to the repository with the new version (GitHub pull requests do not update tags)
```

**Review by xnox — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -27,11 +27,48 @@ TODO: Write documentation for how to build initrd locally
...
+
+1. Compare with the latest package that was uploaded to the snappy-dev
+PPA to make sure that the changes are correct.  For this, you can
+download the .dsc file and the tarball from the PPA, then run debdiff
+to find out the differences:
+
+        dget https://launchpad.net/~snappy-dev/+archive/ubuntu/image/+sourcefiles/ubuntu-core-initramfs/<old_version>/ubuntu-core-initramfs_<old_version>.dsc
+        debdiff ubuntu-core-initramfs_<old_version>.dsc ubuntu-core-initramfs_<new_version>.dsc > diff.txt
+
+1. Upload to the snappy-dev PPA
```
Comment (xnox) [inline README.md:55]:
```suggestion
1. Upload, or request sponsorship, to the snappy-dev PPA
```

**Comment by alfonsosanchezbeato** [comment]
@xnox thanks for the review, I've added now your suggestions to the PR.

File: README.md [hunk 8]
```diff
@@ -27,11 +27,49 @@ TODO: Write documentation for how to build initrd locally
...
 
-TODO: Write documentation for how to integrate initrd.img to your (custom or canonical) kernel 
+1. Update the changelog with latest changes since last release (use `dch -i` for this)
+1. Commit using `debcommit --release -a` which will commit the changelog update & tag the release
+1. Propose a PR to the repo with the new changelog, get it reviewed and merged
+1. Push the tag to the repository with the new version (GitHub pull requests do not update tags)
+1. Build the source package by running
+
+        git clean -ffdx
+        gbp buildpackage -S -sa -d --git-ignore-branch
```
Comment (anonymouse64) [inline README.md:46]: can't you just use `--git-ignore-new` ? also `--ffdx` is destructive if you have unrelated files in the tree, might be nice to call out that this command will delete those files

Comment (alfonsosanchezbeato) [reply README.md:47]: I've added a comment so it is more clear that git clean is destructive. But I think we don't want `--git-ignore-new` here: these are the instruction for releasing the package, not for just building it, so it will be good if we get an error if there are uncommited changes.

Comment (xnox) [reply README.md:47]: @anonymouse64 one must really have a clean tree, if one builds a source package with unrelated files in place, one can trivially end up with them in the source package and like accidentatly have a 10GB large orig.tar.gz which contains some UC20 VM image.

Comment (anonymouse64) [reply README.md:47]: hmm that's not been my experience if you use `--git-ignore-new`, I have unrelated files in my snapd git tree but when I build the snapd deb and upload it we don't get all the random stuff in my git tree locally in the package that is uploaded

but sure in any case as long as it's documented that whoever is building this package needs to be sure that they have unrelated files removed, then this is fine with me

**Review by anonymouse64 — APPROVED** [review]

**Review by xnox — APPROVED** [review]
